app/scripts/lib/metrics.js

```diff
@@ -153,6 +156,46 @@ define(function (require, exports, module) {
       this._clearInactivityFlushTimeout();
     },
 
+    _initializeFlowEvents (options) {
+      const notifier = options.notifier;
+
+      if (notifier) {
+        notifier.on('flow.initialize', () => this._initializeFlowModel(options.sentryMetrics));
```

Comment:
Just to explain the perceived strangeness, it was done at the same time I raised #4639. I'm anti-coupling generally and `sentryMetrics` is something I plant to replace with events. In this case it is used once during the initialisation of the object and then never referred to again. I explicitly didn't want to hold a reference lest it encourage further usage/coupling down the line.

But I'm not wedded to it, I'll change it to how you want.